tests/o2a_libs/test_functions.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for all functions module"""
+
+import unittest
+
+from parameterized import parameterized
+
+import o2a.o2a_libs.functions as functions
+from o2a.o2a_libs.el_wf_functions import _reverse_task_map
+
+
+class TestFunctions(unittest.TestCase):
+    @parameterized.expand(
+        [
+            ("foo", "bar", "foo"),
+            ("foo", "", "foo"),
+            ("foo", None, "foo"),
+            ("", "bar", "bar"),
+            (None, "bar", "bar"),
+            ("", "", ""),
+            (None, None, ""),
+        ]
+    )
+    def test_first_not_null(self, str_one, str_two, expected):
+        self.assertEqual(expected, functions.first_not_null(str_one, str_two))
+
+    @parameterized.expand(
+        [
+            ("foo", "bar", "foo ~ bar"),
```

Comment:
A NIT comment. I think it is better to follow the convention of Unit Tests where expected values are on the left side. I would feel much more comfortable if those parameters here and below follow the same convention mirroring the sequence of arguments in "assertEqual" below. It's not exactly the same case, but for me it feels natural to have "expected" on the left. But maybe others feel differently ?

Reply:
Personally I think that's easier to read it like `arg1, arg2, result` than `result, arg1, arg2`. But I do not insist. @mik-laj @sprzedwojski what are your thoughts?